bundles/org.openhab.binding.valloxmv/src/main/java/org/openhab/binding/valloxmv/internal/ValloxMVHandler.java

```diff
@@ -46,15 +46,15 @@
 public class ValloxMVHandler extends BaseThingHandler {
 
     private final Logger logger = LoggerFactory.getLogger(ValloxMVHandler.class);
-    private @Nullable ScheduledFuture<?> updateTasks;
-    private @Nullable ValloxMVWebSocket valloxSendSocket;
+    private @Nullable ScheduledFuture<?> readDataJob;
+    private @Nullable ValloxMVWebSocket valloxSocket;
     private @Nullable WebSocketClient webSocketClient;
 
     /**
      * Refresh interval in seconds.
      */
-    private int updateInterval;
-    private long lastUpdate;
+    private int readDataInterval;
+    private long nextReadDataTime;
```

Comment:
ScheduledFuture includes a method `getDelay(TimeUnit)` that will tell you how much time until the next execution of a periodic task. Please use that method instead of tracking it manually.

Reply:
I did not notice this method earlier but indeed this is correct way. Took few iterations to make it working correctly.